File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/SqlStatements/DeltaTableConstants/EnergyResultColumnNames.cs
// Copyright 2020 Energinet DataHub A/S
//
// Licensed under the Apache License, Version 2.0 (the "License2");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

using System.Reflection;

namespace Energinet.DataHub.Wholesale.CalculationResults.Infrastructure.SqlStatements.DeltaTableConstants;

public class EnergyResultColumnNames
{
    public const string CalculationId = "calculation_id";
    public const string CalculationExecutionTimeStart = "calculation_execution_time_start";
    public const string CalculationType = "calculation_type";
    public const string CalculationResultId = "calculation_result_id";
    public const string GridArea = "grid_area_code";
    public const string EnergySupplierId = "energy_supplier_id";
    public const string Time = "time";
    public const string Quantity = "quantity";

    public const string BalanceResponsibleId = "balance_responsible_id";
    public const string QuantityQualities = "quantity_qualities";
    public const string AggregationLevel = "aggregation_level";
    public const string TimeSeriesType = "time_series_type";
    public const string NeighborGridArea = "neighbor_grid_area_code";
    public const string MeteringPointId = "metering_point_id";
    public const string Resolution = "resolution";

    public static IReadOnlyCollection<string> GetAllNames()
    {
        var fieldInfos = typeof(EnergyResultColumnNames).GetFields(BindingFlags.Public | BindingFlags.Static);
        return fieldInfos.Select(x => x.GetValue(null)).Cast<string>().ToList();
    }

    public static string GetType(string columnName) =>
         columnName switch
         {
             GridArea => "string",
             BalanceResponsibleId => "string",
             CalculationResultId => "string",
             EnergySupplierId => "string",
             Time => "timestamp",
             QuantityQualities => "array<string>",
             AggregationLevel => "string",
             TimeSeriesType => "string",
             CalculationId => "string",
             CalculationType => "string",
             CalculationExecutionTimeStart => "timestamp",
             NeighborGridArea => "string",
             Quantity => "decimal(18,3)",
             MeteringPointId => "string",
             Resolution => "string",
             _ => throw new ArgumentOutOfRangeException(nameof(columnName), actualValue: columnName, "Unexpected column name."),
         };
}

File: source/databricks/calculation_engine/package/databases/wholesale_results_internal/energy_result_column_names.py
# Copyright 2020 Energinet DataHub A/S
#
# Licensed under the Apache License, Version 2.0 (the "License2");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.


from package.databases.wholesale_results_internal.result_column_names import (
    ResultColumnNames,
)


class EnergyResultColumnNames(ResultColumnNames):
    """Column names for the energy result storage model."""

    time_series_type = "time_series_type"
    neighbor_grid_area_code = "neighbor_grid_area_code"
    """The delta table column name ought to be renamed to 'from_grid_area'.
    It is, however, rather complicated compared to the value unless using Delta Lake column mapping,
    which comes with its own set of problems/limitations."""
    balance_responsible_id = "balance_responsible_id"
    """Obsolete"""
    aggregation_level = "aggregation_level"
    """Obsolete"""
    metering_point_type = "metering_point_type"
    metering_point_id = "metering_point_id"
    quantity = "quantity"
    quantity_qualities = "quantity_qualities"
    resolution = "resolution"